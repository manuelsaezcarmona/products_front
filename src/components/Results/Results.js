import React from 'react';
import { CloseOutlined } from '@ant-design/icons';
import styles from './styles.module.scss';
import { selectResultsSearch } from '../../redux/selectors';

export default function Results() {
  const { itemsCount, searchTerm } = selectResultsSearch();

  const handleCloseClick = () => {
    console.log('Cierra la ventana');
  };

  return (
    <div className={styles.results}>
      <p className={styles.results__content}>{itemsCount} Items found</p>
      <h2 className={styles.results__title}>Search Results</h2>
      <h2 className={styles.results__title}>for {searchTerm}</h2>
      <CloseOutlined
        className={styles.results__closeIcon}
        onClick={handleCloseClick}
      />
    </div>
  );
}
