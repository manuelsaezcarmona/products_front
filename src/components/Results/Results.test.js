import { render, screen } from '@testing-library/react';
import { Provider, useSelector } from 'react-redux';
import thunk from 'redux-thunk';
import { BrowserRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import {
  selectResultsSearch,
  selectVisibilityResults
} from '../../redux/selectors';
import { startGetAllProducts } from '../../redux/actions/products.action';
import Results from './Results';

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

const initialState = {
  products: {
    products: [],
    productAdded: {
      productName: '',
      description: '',
      imageURL: '',
      isFavourite: false,
      price: 0,
      section: ''
    },
    searchResults: {
      itemsCount: 3,
      searchTerm: 'algo'
    }
  },
  ui: {
    currentPage: 1,
    currentFilter: '',
    currentOrder: '',
    resultsVisible: true
  }
};

const store = mockStore(initialState);

store.dispatch = jest.fn();

jest.mock('react-redux', () => ({
  useSelector: jest.fn().mockImplementation((selector) => selector())
}));

jest.mock('../../redux/selectors', () => ({
  selectResultsSearch: jest.fn().mockReturnValue({
    searchResults: {
      itemsCount: 3,
      searchTerm: 'algo'
    }
  }),
  selectVisibilityResults: jest.fn().mockReturnValue({
    ui: {
      resultsVisible: true
    }
  })
}));

describe('Given the Header Component', () => {
  beforeEach(() => {
    render(
      <Provider store={store}>
        <BrowserRouter>
          <Results />
        </BrowserRouter>
      </Provider>
    );
  });
  test('The component is rendered (take a snapshot)', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <Results />
      </Provider>
    );
    expect(asFragment(<Results />)).toMatchSnapshot();
  });
});
