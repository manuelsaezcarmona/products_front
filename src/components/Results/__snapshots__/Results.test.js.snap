// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given the Header Component The component is rendered (take a snapshot) 1`] = `<DocumentFragment />`;
